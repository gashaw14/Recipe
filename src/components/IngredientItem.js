import React from 'react';

const IngredientItem =({ingredient, checkbox})=>(
    <div className='checbox'>
      {ingredient && <label><input type='checkbox' />{ingredient}</label> } 
    </div>
)
export default IngredientItem;