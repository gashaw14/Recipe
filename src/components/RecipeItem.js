import React from 'react';
import {Link} from 'react-router-dom';


const RecipeItem =({id,menuTitle,checkbox})=>(
    <div>
    <button className='big-button'>
    <Link to={`/edit/${id}`}>
    <h3>{menuTitle}</h3>
    <p>{checkbox && 'It has Ingredien'}</p>
    </Link>
    </button>
   
    </div>
)

export default RecipeItem;