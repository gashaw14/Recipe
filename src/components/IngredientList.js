import React from 'react';
import IngredientItem from './IngredientItem';
import selectRecipe from '../selctors/recipe';
import { connect } from 'react-redux';

const IngredientList = (props)=>(
<div>
 {props.recipes.map(recipe=>{
     return <IngredientItem  key={recipe.id} {...recipe} />
     }
     
     )}
     
     
</div>
)
  

const mapStateToProps = (state)=>{
 return{
     recipes: selectRecipe(state.recipes, state.filters)
 }
}

export default connect(mapStateToProps)(IngredientList)