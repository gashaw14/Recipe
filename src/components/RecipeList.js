import React from 'react';
import RecipeItem from './RecipeItem';
import selectRecipe from '../selctors/recipe';
import { connect } from 'react-redux';

export const RecipeList = (props)=>(
    props.recipes.length === 0?
    (<p>no recipes</p>):(
      props.recipes.map(recipe=>{
    return <RecipeItem key={recipe.id} {...recipe}/> }) 
    )
)

const mapStateToProps = (state)=>{
 return{
     recipes: selectRecipe(state.recipes, state.filters)
 }
}

export default connect(mapStateToProps)(RecipeList)