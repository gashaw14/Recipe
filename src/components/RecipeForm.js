import React from 'react';
import IngredientList from './IngredientList';

export default class RecipeForm extends React.Component{

constructor(props){
super(props);
this.state={
    menuTitle: props.recipe ? props.recipe.menuTitle:'',
    steps: props.recipe ? props.recipe.steps:'',
    ingredient: props.recipe ? props.recipe.ingredient:'',
    checkbox:props.recipe ? props.recipe.checkbox: false,
    error:''
}

}
onMenuTitleChange=(e)=>{
 const menuTitle = e.target.value
 this.setState(()=>({menuTitle}))
}

onStepsChange =(e)=>{
 const steps = e.target.value
 this.setState(()=>({steps}))
}
onIngredientChange=(e)=>{
const ingredient = e.target.value;
this.setState(()=>({ingredient}));
}

handleSubmit= (e)=>{
e.preventDefault()
if(!this.state.menuTitle ){
this.setState(()=>({error:'please enter menu title'}))
} else{
    this.setState(()=>({errors:''}))
    this.props.onSubmit({
        menuTitle: this.state.menuTitle,
        steps: this.state.steps,
        ingredient: this.state.ingredient,
        checkbox: !this.state.checkbox
    })
}
}

render(){
    return(
    <div className='form'>
      {this.state.error && <p>{this.state.error}</p>}
     <form onSubmit={this.handleSubmit}>
         <input
         type='text'
         placeholder='menu name'
         value= {this.state.menuTitle}
         onChange ={this.onMenuTitleChange}
         ></input>
         <textarea
         type='text'
         placeholder='cooking steps'
         value={this.state.steps}
         onChange={this.onStepsChange}
         ></textarea>
        <h1>Ingredients</h1>
        <IngredientList />
        <input 
        type='text' 
        placeholder='add ingredient'
        value={this.state.ingredient}
        onChange={this.onIngredientChange}
        ></input>
        <button >Add Ingredient</button>
     </form>   
     
    </div>
    )
}
}