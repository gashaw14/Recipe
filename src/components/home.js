import React from 'react';
import RecipeList from './RecipeList';
import {Link} from 'react-router-dom';
import RecipeListFilter from './RecipeListFilter';

const Home =()=>{
    return (
        <div>
         <RecipeListFilter />
         <RecipeList /> 
         <button>
          <Link to='/create'>Add Recaipe</Link>
         </button>  
        </div>
    )
       
    
}

export default Home;