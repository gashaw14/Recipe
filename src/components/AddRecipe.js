import React from 'react';
import RecipeForm from './RecipeForm';
import { addRecipe } from '../actions/recipe';
import {connect} from 'react-redux';

export class AddRecipePage extends React.Component{
  onSubmit=(recipe)=>{
    this.props.addRecipe(recipe);
    this.props.history.push('/');
  }
  render(){
    return(
      <div>
      <h1>Add Recipe</h1>
      <RecipeForm
        onSubmit={this.onSubmit}
      />
    </div>
    )
  }
}

  const mapDispatchToProps =(dispatch)=>{
    return{
      addRecipe:(recipe)=> dispatch(addRecipe(recipe))
    }
  }
  export default connect(undefined,mapDispatchToProps)(AddRecipePage);
  