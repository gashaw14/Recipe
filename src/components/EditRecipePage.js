import React from 'react';
import { connect } from 'react-redux';
import RecipeForm from './RecipeForm';
import {editRecipe, removeRecipe} from '../actions/recipe';

export class EditRecipePage extends React.Component{
  onSubmit=(recipe)=>{
    this.props.editRecipe(this.props.recipe.id, recipe);
    this.props.history.push('/');

  }
  onClick=()=>{
    this.props.removeRecipe({id:this.props.recipe.id}) 
    this.props.history.push('/');  
  }
    render(){
        return(
            <div>
         <RecipeForm 
              recipe ={this.props.recipe}
              onSubmit={this.onSubmit}
          />  
          <button
          onClick={this.onClick}
          >remove</button>
            </div>
         
        )
    }
}

const mapDispatchToProps = (dispatch)=>{
    return{
        editRecipe:(id,recipe)=> dispatch(editRecipe(id, recipe)),
        removeRecipe:(data)=>dispatch(removeRecipe(data))
    }
}
const mapStateToProps = (state,props)=>{
return{
    recipe:state.recipes.find((recipe)=> recipe.id === props.match.params.id)
}
}

export default connect(mapStateToProps,mapDispatchToProps)(EditRecipePage);