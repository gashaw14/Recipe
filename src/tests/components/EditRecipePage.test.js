import React from 'react';
import {shallow, ReactWrapper} from 'enzyme';
import { EditRecipePage} from '../../components/EditRecipePage';
import toJSON from 'enzyme-to-json';

let editRecipe,removeRecipe, history, wrapper;
const recipes =[{
    id:'1',
    menuTitle:'VEGI', 
    steps:'ttt',
    ingredient:'aaa',
    checkbox:false

},{
   id:'2',
    menuTitle:'meat', 
    steps:'cook',
    ingredient:'yyyy',
    checkbox:false
}
]

beforeEach(()=>{
    editRecipe = jest.fn();
    removeRecipe = jest.fn();
    history = {push:jest.fn()};
    wrapper = shallow(<EditRecipePage 
    editRecipe={editRecipe} 
    removeRecipe={removeRecipe} 
    history={history}
    recipe = {recipes[0]}
    />);
    
})
test('test render edit recipe component',()=>{
expect(toJSON(wrapper)).toMatchSnapshot()

})
test('test handle onSubmit',()=>{
    wrapper.find('RecipeForm').prop('onSubmit')(recipes[0]);
    expect(history.push).toHaveBeenCalledWith('/');
    expect(editRecipe).toHaveBeenCalledWith(recipes[0].id,recipes[0] );

})
test('test handle removeRecipe', () => {
  wrapper.find('button').simulate('click');
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(removeRecipe).toHaveBeenLastCalledWith({
    id: recipes[0].id
  });
});
