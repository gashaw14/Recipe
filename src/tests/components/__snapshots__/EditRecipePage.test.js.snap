// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render edit recipe component 1`] = `
<div>
  <RecipeForm
    onSubmit={[Function]}
    recipe={
      Object {
        "checkbox": false,
        "id": "1",
        "ingredient": "aaa",
        "menuTitle": "VEGI",
        "steps": "ttt",
      }
    }
  />
  <button
    onClick={[Function]}
  >
    remove
  </button>
</div>
`;
